### PR TITLE
Leaf 3818 - connect button to new api

### PR DESCRIPTION
**Pull request by jampaul3**

### CODE REVIEW

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

File: LEAF_Nexus/sources/Employee.php [hunk 1]
```diff
@@ -171,7 +171,7 @@ private function formatUserNames(array $employee_list): array
     {
         $local_employee_usernames = [];
 
-        foreach ($employee_list as $employee) {
+        foreach ($employee_list['data'] as $employee) {
```
Comment (shaneodd) [inline LEAF_Nexus/sources/Employee.php:174]: Should 'data' be checked if it exists and is array?

**Review by shaneodd — CHANGES_REQUESTED** [review]

File: LEAF_Nexus/sources/Employee.php (hunk 1 above)
Comment (raphaelfretter) [reply LEAF_Nexus/sources/Employee.php:174]: The employee_list appears to be checked if empty before calling this method

Comment (shaneodd) [reply LEAF_Nexus/sources/Employee.php:174]: employee_list  is checked but the ['data'] would not be checked. We would know employee_list  is an array at this point however what is ['data']

Comment (jampaul3) [reply LEAF_Nexus/sources/Employee.php:174]: In an attempt to get the json responses in the format that Michael Shaffer is looking for them in data is just that, it's the data being sent back. The json response has 2 elements to it, the status and the data. status just gives the status with a code and message (the message is generally blank when the status is ok,), the  data is usually going to be an array of the data being returned.

**Pull request merged**